### PR TITLE
[REG-2001] Overhaul Validations Docs

### PR DESCRIPTION
**Pull request by abeizer**
* Combined validations content into one doc
  * Improves the look of the side-nav
  * Simplifies the url for the page
* Reorganized sections within the Validations page
* Filled in a lot of missing information / clarifications
* Added a couple gifs + screenshots
* Rules and Steps have neat little tables now

https://www.loom.com/share/595f290c8ce64de18d25310720773fd5

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: docs/core-concepts/validation-suites.mdx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+![Create a Validation Suite](img/validation-suites/create-validation-suite.gif)
+
+## Validation Scenarios
+
+A **Validation Scenario** defines a functional test case by asserting that specific actions are taken during
+a Gameplay Session and that they produce expected changes in the game state.
+Steps can assert that a game scene is loaded, wait for mouse and keyboard inputs, check for the existence
+of specific game objects, and much more.
+
+Validation Scenarios execute in the same order as they appear in the Validation Suite dashboard, but order does not impact how Scenarios run.
```
Comment (batu) [inline docs/core-concepts/validation-suites.mdx:30]: This is the first part that I felt confused. If scenarios execute in the order, then in my mind the order DOES impact how scenarios are run? 

Is it that the order doesn't impact whether the scenarios run? If this is not it, I don't think the text conveyed to me in how they don't impact one another. (or is it that because they run sequentially, they don't interfere with one another?)

Comment (batu) [reply docs/core-concepts/validation-suites.mdx:30]: Oh... I see, this is about multiple Scenarios, not the steps in the scenarios. To clear this up consider adding a priming sentence that goes something like "You can have multiple scenarios in ..."

File: docs/core-concepts/validation-suites.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+Each Scenario starts at the first tick of the Gameplay Session, and a Suite continues to run remaining Scenarios when one fails.
+
+### Creating a Validation Scenario
+
+Open your Suite in the [Validation Suites dashboard](https://play.regression.gg/validations) and click the **Create Scenario** button.
+This opens the **Scenario Builder**, where you can view game state data and a screenshot for each tick in the Gameplay Session,
+and use this information to build steps for your Scenario.
+
+![Screenshot of a scenario](img/validation-suites/scenario-example.png)
```
Comment (batu) [inline docs/core-concepts/validation-suites.mdx:39]: I think this is mostly self explanatory but at this point in the doc the reader doesn't know what each of the lines does. Having a sentence that explains what this scenario accomplishes can help them ground their assumptions in what it actually does, and potentially how it adds value. 

Maybe something like:
This Scenario does, x,y,z and can be used to ensure that a,b,c doesn't break as we make changes to our game.

File: docs/core-concepts/validation-suites.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+### Validation Steps
+
+Validation Steps execute in the same order as they are defined in the Scenario Builder, from the top down.
+When a step fails, the Scenario fails and ceases execution.
+
+Many steps progress through the Gameplay Session until a tick satisfies their criteria.
+These steps are prefixed with the words **Wait for**, and they tell the next step in the Scenario to execute from that tick.
+In this way, steps run sequentially through the Session and can be used to establish cause-and-effect relationships between game state changes.
+If a "Wait for" step reaches the end of the Gameplay Session without encountering a tick that satisfies its criteria, then that step will fail.
```
Comment (batu) [inline docs/core-concepts/validation-suites.mdx:49]: I find myself nodding along saying 'yees yeees' but not really getting it. A worked out running example might be helpful, here or at the end.

**Review by batu — COMMENTED** [review]
I added a few line comments dumping my mental process as I was going through the document. I think the prose is good, but the organization can be made better by grounding everything with an example at the start.

The problem is that there are many components that the learner needs to understand: Validation, scenarios, rules, steps etc. When these are taught to the user one at a time, there is not much context initially for them to land relatively well. Starting with an example will help this. Some call this ["teaching the whole game"](https://www.fast.ai/posts/2016-10-08-teaching-philosophy.html#good-education-starts-with-the-whole-game).

**Review by addisonbgross — COMMENTED** [review]
Overall this reads well and is nice and concise. I think @batu 's point about having a explained Scenario example is a great idea!

**Review by vontell — APPROVED** [review]

**Review by batu — APPROVED** [review]
Forgot to approve this, if we choose to add a grounding example that can be a future PR

**Comment by abeizer** [comment]
@batu @addisonbgross  I liked the suggestion of adding an explanation for the example so here's the updated section:

image

File: docs/core-concepts/validation-suites.mdx (hunk 3 above)
Comment (abeizer) [reply docs/core-concepts/validation-suites.mdx:63]: Let me know whether the example scenario helps make sense of this text, or if I need to rework this to give the user something more concrete to hook into

File: docs/core-concepts/validation-suites.mdx (hunk 1 above)
Comment (abeizer) [reply docs/core-concepts/validation-suites.mdx]: Added that primer sentence and clarified that scenarios won't impact each other

**Review by addisonbgross — APPROVED** [review]
Premium overhaul 👍🏻

**Review by batu — APPROVED** [review]
Sweet!
